src/api/app/controllers/application_controller.rb

```diff
@@ -89,6 +89,13 @@ def self.validate_action(opt)
 
   protected
 
+  def feature_enabled?(feature)
+    return if Flipper.enabled?(feature.to_sym, User.possibly_nobody) || Flipper.enabled?(feature.to_sym)
+    render file: Rails.root.join('public/404'), status: :not_found, layout: false
+  end
+
+  # TODO: remove when all the migration from Feature to Flipper is finished.
+  # It'll be replaced by feature_enabled?
   def feature_active?(feature)
     return if Feature.active?(feature)
     render file: Rails.root.join('public/404'), status: :not_found, layout: false
```

Comment:
why it cannot return simply boolean without render?

Reply:
@vpereira, We use this method as a `before_action` into different controllers like `image_template` to make them not accessible.

Reply:
Exactly, useful when someone goes directly to a URL that only exists if the feature is active. But if the user is not allowed to access to the feature, he/she will see the 404.